### PR TITLE
Improve eslint

### PR DESCRIPTION
**Pull request by atrovato**
Based on #155 comment: split PR eslint first :)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -13,12 +13,13 @@ module.exports = {
...
   rules: {
+    "space-before-function-paren": ["error", "always"],
     "no-unused-vars": [
       "error",
       {
         args: "none"
       }
     ],
-    "semi": "error"
+    semi: "error"
```
Comment (atrovato) [inline .eslintrc.js:23]: @rzr you commented:
> please explain why this was changed ?

As file content is json, quotes are not required here.

File: package.json [hunk 2]
```diff
@@ -59,7 +59,8 @@
     "ws": "^7.4.2"
   },
   "scripts": {
-    "lint": "eslint '**/*.js'",
+    "lint": "eslint \"**/*.js\"",
+    "lint-fix": "eslint \"**/*.js\" --fix",
```
Comment (atrovato) [inline package.json:63]: I use `\"` to be able to run on windows env.

**Comment by rzr** [comment]
Please add a change to ignore modules too ie :

https://github.com/abandonware/noble/pull/171/checks?check_run_id=1844386700# 

```
s/home/runner/work/noble/noble/node_modules/mocha/lib/mocha.js:1215for await (const fixtureFn of fixtureFns) {^^^^^
```

**Comment by atrovato** [comment]
This error is not linked to eslint but to node 8 build. Should I do in this PR or in a separate one?

**Comment by atrovato** [comment]
With eslint 7, node 8 is no longer supported

https://eslint.org/docs/user-guide/migrating-to-7.0.0#nodejs-8-is-no-longer-supported

**Comment by atrovato** [comment]
And `As of v8.0.0, Mocha requires Node.js v10.12.0 or newer.`

https://mochajs.org/#installation

**Comment by atrovato** [comment]
Ok I changed `nodepackage.yml` to ignore lin on node 8, and to use mocha 7.2.0 on node 8.

**Comment by rzr** [comment]
sounds good to me

**Review by rzr — CHANGES_REQUESTED** [review]
please open new pr with this change : https://github.com/abandonware/noble/pull/171/commits/b76fa34ab1b8e27299b49eb29ae2741da01cc5d5#diff-b29da93bc686f97c01ba07c233850a8c2a21d53cbe9bfc153f905640106da66b and then your eslint fixes then you can rebase everything to make sure nothing break

**Review by rzr — CHANGES_REQUESTED** [review]
please rebase with only eslint changes

**Comment by atrovato** [comment]
Done :)

**Review by rzr — APPROVED** [review]

**Pull request merged**